### DISCUSSION TITLE
Problems with push for ios

### DISCUSSION THREAD

**Discussion by 1amfine2333** [dani-garcia/vaultwarden#3893] [Q&A]
I need help, I have enabled PUSH_INSTALLATION_ID and PUSH_INSTALLATION_KEY,but I still can't receive the push.
docker-compose.yml
<img width="790" alt="image" src="https://github.com/dani-garcia/vaultwarden/assets/24676292/68cd9c1c-8af2-469e-89d4-36bbb3fc2018">
logs
<img width="1256" alt="image" src="https://github.com/dani-garcia/vaultwarden/assets/24676292/258fc295-effd-43a8-959b-f624bfca451b">

**Comment by BlackDex** [comment]
You generated a EU key and not a US maybe?
Vaultwarden currently only supports US.

**Reply by 1amfine2333** [reply]
I generated a US key.

Is there any way to verify whether the key is available?

**Reply by BlackDex** [reply]
I see it says timeout. Kinda looks like there is no internet connection from the container.

Could you generate a support string via the admin interface?

**Reply by 1amfine2333** [reply]
Wait a minute, I closed the admin interface before

**Comment by 1amfine2333** [comment]
> I see it says timeout. Kinda looks like there is no internet connection from the container.
> 
> Could you generate a support string via the admin interface?

I opened the admin interface. What should I do?

<img width="892" alt="image" src="https://github.com/dani-garcia/vaultwarden/assets/24676292/a22f43eb-43cb-4f93-a78b-408320acd67c">

**Reply by BlackDex** [reply]
Generate a support string via the diagnostics page

**Reply by 1amfine2333** [reply]
ok, I think it's Internet access, but my cloud server has Internet access
<img width="1031" alt="image" src="https://github.com/dani-garcia/vaultwarden/assets/24676292/39f79a5f-7cdd-41aa-afd8-e6ef92fd68c0">

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.29.2
* Web-vault version: v2023.7.1
* OS/Arch: linux/x86_64
* Running within Docker: true (Base: Debian)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: false
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: n/a
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.41.2
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 


```json
{
  "_duo_akey": null,
  "_enable_duo": true,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": true,
  "disable_icon_download": false,
  "domain": "*****://***********************",
  "domain_origin": "*****://***********************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": false,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 600000,
  "push_enabled": true,
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": false,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": null,
  "smtp_password": null,
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

**Reply by BlackDex** [reply]
It seems to be able to do dns requests, but not access to internet. At least not from within container.

**Reply by 1amfine2333** [reply]
But can really surf the Internet.

I want to set up a proxy.

**Reply by 1amfine2333** [reply]
OK, I found the problem. I added an proxy and push it normally.

**Reply by 1amfine2333** [reply]
Thanks ！！！

**Comment by 1amfine2333** [comment]
I found a new problem, but I see that it has been fixed. #3885
I don't know if there is no update on the docker hub.
<img width="1264" alt="image" src="https://github.com/dani-garcia/vaultwarden/assets/24676292/16a56f2a-ab1d-4e68-98ac-37bcb42dd759">
<img width="779" alt="image" src="https://github.com/dani-garcia/vaultwarden/assets/24676292/3e926372-f4dd-4e7b-943c-5d38d0a5c2e2">

**Reply by stefan0xC** [reply]
If you use `testing` the fix is included (`testing` is always build off the current `main` branch). Otherwise you'll have to wait for the next release.

**Reply by 1amfine2333** [reply]
Thank you. Then I'll use testing first and update it when the main branch comes out.
